state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1301,37 +1403,19 @@ pub mod pallet {
 		#[pallet::weight(T::WeightInfo::vault_swap_request())]
 		pub fn vault_swap_request(
 			origin: OriginFor<T>,
-			input_asset: TargetChainAsset<T, I>,
-			output_asset: Asset,
-			deposit_amount: <T::TargetChain as Chain>::ChainAmount,
-			destination_address: EncodedAddress,
-			deposit_metadata: Option<CcmDepositMetadata>,
-			tx_id: TransactionInIdFor<T, I>,
-			deposit_details: Box<<T::TargetChain as Chain>::DepositDetails>,
-			broker_fee: Beneficiary<T::AccountId>,
-			affiliate_fees: Affiliates<AffiliateShortId>,
-			refund_params: Box<ChannelRefundParameters>,
-			dca_params: Option<DcaParameters>,
-			boost_fee: BasisPoints,
+			block_height: TargetChainBlockNumber<T, I>,
+			deposits: Vec<VaultDepositWitness<T, I>>,
```

Comment:
I imagine the benefit is to be able to submit multiple Vault swaps in the same extrinsic, so potentially decreasing the numbers of extrinsics. That improvement is probably irrelevant for any chain besides BTC as we most likely won't have many if this instances.
On the other hand, I do see some room for this being problematic. My understanding is that to have consensus well need engines to submit the exact same extrinsic. In Solana where we can only query the tip of the chain, submitting all Vault swaps as a vector will be problematic:
- Start at block 0
- At block 10 a Vault swap is executed
- At block 15 a Vault Swap is executed
- Some engines query between 10 and 11 submitting the extrinsic with one Vault Swap and then later on another one with the second Vault swap.
- Other engines query at block 16 submitting the extrinsic with two Vault swaps.
Then we wouldn't reach consensus.
We can choose to have the engine handle that (e.g in Solana never submit a vector) but it could be prone to error in the future. I'm fine with keeping it like this, since there is some upside to this, but I can see the argument of aiming for a more safe approach. I just wanted to highlight it and also to check that my understanding is correct, especially since we'll also have to update this for Solana Vault swaps.

Reply:
Fwiw I think this is only problematic for Solana atm but I wouldn't be surprised if other fast chains we might integrate in the future would also fall into that category.

Reply:
For solana it's fine, the way the elections are implemented is that it destructures them to process each deposit independently, regardless of how they're submitted e.g.
validator 1 submits [1, 2, 3], validator 2 submits [1, 2] and validator 3 submits [2], then [1 ,2] go through since they both reached 2/3. 

For bitcoin we witness for a particular block so every engine should agree.

Reply:
yes, thats correct.